src/Compilers/Core/Portable/DiagnosticAnalyzer/DefaultAnalyzerAssemblyLoader.Core.cs

```diff
@@ -70,6 +70,9 @@ internal class DefaultAnalyzerAssemblyLoader : AnalyzerAssemblyLoader
                "System.Xml.XDocument",
                "System.Xml.XPath.XDocument");
 
+        // This is the default context where compiler (and some of its dependencies) are being loaded into, which might be differernt from AssemblyLoadContext.Default.
+        private static readonly AssemblyLoadContext s_compilerLoadContext = AssemblyLoadContext.GetLoadContext(typeof(DefaultAnalyzerAssemblyLoader).GetTypeInfo().Assembly)!;
```

Comment:
It's not clear to me how the 'static readonly' could lead to problems, given that there's 1 static field per assembly, essentially, but I think @jaredpar's suggestion was to eliminate the static field at the same time as introducing an instance field to the `DirectoryLoadContext`.

Reply:
I sure can change it to an instance field, but it'd be great if someone can show me what kind of problem it might cause :)

Reply:
From the [design note](https://github.com/dotnet/coreclr/blob/master/Documentation/design-docs/assemblyloadcontext.md), it is stated that statics are tied to context, so this shouldn't be a problem it seems

Reply:
All I would ask is that we do either the static field or the instance field not both. However this can happen in a follow up.

Reply:
That makes sense.